src/transactions/TransactionFrame.cpp

```diff
@@ -1130,10 +1162,14 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
             outerMeta.pushOperationMetas(std::move(operationMetas));
             outerMeta.pushTxChangesAfter(std::move(changesAfter));
 #ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
-            // ContractEvents should only occur in InvokeHostFunctionOps which
-            // should only occur in txs with a single op. This is enforced in
-            // InvokeHostFunctionOpFrame::doCheckValid but re-assert here.
-            releaseAssertOrThrow(mEvents.empty() || mOperations.size() == 1);
+            // The events should have an OperationEvents for each smart op, and
+            // since smart ops can't be mixed with classic ops, mEvents should
+            // be empty if this tx has a classic op.
+            if (mOperations.size() == mEvents.size() &&
```

Comment:
Condition does not look like it's matching the intent?

Condition should be
```
auto isSmartTx = mOperations.at(0)->isSmartOperation();
if (isSmartTx && mOperations.size() != mEvents.size()) ||
(!isSmartTx &&  !mEvents.empty())) {
...
```

I don't see how it can get simplified to what is here.
If you change the code, you may want to remove the comment as it would be self-explanatory.

Reply:
The comment wasn't right - I just wanted to check that there was an event vec per operation for smart. But checking that classic has no events is a good idea too. Updated.